### PR TITLE
nixos/jackline: init

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
The new services.jackline module allows using the already packaged
jackline XMPP client in an interactive screen session. The module's
design is heavily inspired by the weechat module, which also provides a
screen session, just for weechat.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/jackline.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      description = ''
+        Whether to enable jackline, a minimalistic secure XMPP client in OCaml.
+
+        By doing so, jackline will be run in a screen session. One can attach
+        this session via <literal>sudo -u jackline screen -x jackline-screen</literal>.
+      '';
+      type = types.bool;
+      default = false;
+      example = true;
+    };
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/jackline.nix:19]:
```suggestion
    enable = mkEnableOption ''
        Whether to enable jackline, a minimalistic secure XMPP client in OCaml.
        
        By doing so, jackline will be run in a screen session. One can attach
        this session via <literal>sudo -u jackline screen -x jackline-screen</literal>.
      '';
```

File: nixos/modules/services/misc/jackline.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    security.wrappers.screen = {
+      setuid = true;
+      owner = "root";
+      group = "root";
+      source = "${pkgs.screen}/bin/screen";
+    };
+
+    systemd.services.jackline = {
+      description = "jackline, a minimalistic secure XMPP client in OCaml";
```
Comment (legendofmiracles) [inline nixos/modules/services/misc/jackline.nix:61]:
```suggestion
      description = "Jackline, a minimalistic secure XMPP client in OCaml";
```

File: nixos/modules/services/misc/jackline.nix (hunk 1 above)
Comment (oxzi) [reply nixos/modules/services/misc/jackline.nix:19]: Your suggested change would result in a description of "Whether to enable Whether to enable jackline, a minimalistic secure XMPP client in OCaml. […] </literal>.."

https://github.com/NixOS/nixpkgs/blob/a26de40d124cbe084aef15e92746bd26ef7b4e51/lib/options.nix#L93-L100

I have chosen not to use `mkEnableOption`, as it is imho not really usable for multi-line descriptions.

File: nixos/modules/services/misc/jackline.nix (hunk 2 above)
Comment (oxzi) [reply nixos/modules/services/misc/jackline.nix:61]: Seriously? The program's name is always written in lower cases within this file. There is literally no need to change this here.

**Comment by oxzi** [comment]
Closed as this PR is stalled over half a year and I stopped using jackline as the peer group decided to move to Matrix..
Anyone, please feel free to create a new PR based on this one.

**Pull request closed without merging**